tsdb/head.go

```diff
@@ -1177,6 +1190,7 @@ func (h *Head) gc() {
 
 	h.symbols = symbols
 	h.values = values
+	h.metrics.labelpairs.Set(float64(len(h.values)))
```

Comment:
And this would be the number of label names.

You'd need to iterate over h.values to count up the number of unique label pairs.